app/models/email_address.rb

```diff
@@ -33,6 +33,15 @@ def confirmation_period_expired?
     Time.zone.now > expiration_time
   end
 
+  def domain
+    email&.split('@')&.last
```

Comment:
IMO clearer to use the mail gem to parse (we use this same pattern elsewhere in the codebase)
```suggestion
    email && Mail::Address.new(email).domain
```